src/pyroboplan/core/utils.py

```diff
@@ -356,3 +372,78 @@ def set_collisions(model, collision_model, body1, body2, enable):
                 collision_model.addCollisionPair(pair)
             else:
                 collision_model.removeCollisionPair(pair)
+
+
+def calculate_collision_vector_and_jacobians(
+    model, collision_model, data, collision_data, pair_idx, q
+):
+    """
+    Given collision and distance results from collision model data, computes the collision vector
+    and collision Jacobians at both collision points.
+
+    This is useful for algorithms that perform collision avoidance, such as IK and trajectory optimization.
+
+    Note that forward kinematics, collision, and distance checks must be evaluated first to populate the
```

Comment:
Will this raise an exception if they're missing?

Reply:
nope. This is sadly just how Pinocchio works, that the model and collision data needs to be initialized or else it has all default values.